src/easylink/pipeline_graph.py

```diff
@@ -114,3 +196,67 @@ def get_input_output_files(self, node: str) -> Tuple[List[str], List[str]]:
     def spark_is_required(self) -> bool:
         """Check if the pipeline requires spark resources."""
         return any([implementation.requires_spark for implementation in self.implementations])
+
+    def validate_edge_params(self, edge_params: list[EdgeParams]) -> None:
+        # Group edges by output slot to check for conflicts
+        input_slot_groups = defaultdict(list)
+
+        for edge in edge_params:
+            input_slot_groups[edge.input_slot].append(edge)
+
+        # Check each group of edges sharing the same output slot
+        for input_slot, edges in input_slot_groups.items():
+            if len(edges) <= 1:
+                continue
+
+            # Check for conflicting input slots
+            output_slot_names = set(edge.output_slot for edge in edges)
+            if len(output_slot_names) > 1:
+                conflicting_edges = [
+                    f"({edge.source_node}->{edge.target_node}, output={edge.output_slot})"
+                    for edge in edges
+                ]
+                raise ValueError(
+                    f"Input slot '{input_slot}' is connected to different output slots in edges: {', '.join(conflicting_edges)}"
+                )
+
+            edge_final_steps = set(
+                self.nodes[edge.source_node]["implementation"].metadata_steps[-1]
+                for edge in edges
+            )
+            if len(edge_final_steps) > 1:
+                raise ValueError(
+                    f"Input slot '{input_slot}' is connected to nodes that implement different schema steps"
+                )
```

Comment:
The concern is if we had (presumably unintendedly) two input slots that shared a name /env var / validator but were connected to two different steps. Maybe that's overkill, especially since we will explicitly be de-duplicating these slots later.